cmp/report_text.go

```diff
@@ -0,0 +1,382 @@
+// Copyright 2019, The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE.md file.
+
+package cmp
+
+import (
+	"bytes"
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/google/go-cmp/cmp/internal/flags"
+)
+
+var randBool = rand.New(rand.NewSource(time.Now().Unix())).Intn(2) == 0
+
+type indentMode int
+
+func (n indentMode) appendIndent(b []byte, d diffMode) []byte {
+	if flags.Deterministic || randBool {
```

Comment:
Why?

I was using the output of `cmp.Diff` in a Go runnable example, and it was failing intermittently. I'd like to understand why do this.

(Sorry, not sure whether it's OK to comment in an old pull request... I got here looking for rationale.)

Reply:
The output of `cmp.Diff` is not defined. It can and will change over time. For example, this CL completely changed the output; the new output is better, but the change breaks any tests which depended on the old one.

Making the output trivially nondeterministic ensures that anyone depending on stable output from `cmp.Diff` will discover the problem immediately rather than at some future point in time when the output changes in a more significant way.

Reply:
@dneil is correct. I mailed out #145 to document this.